graph/repo.go

```diff
@@ -33,6 +33,14 @@ func TryMakeURI(cloneURL string) (string, error) {
 	url, err := url.Parse(cloneURL)
 	if err != nil {
 		return "", fmt.Errorf("MakeURI(%q): %s", cloneURL, err)
+	} else if url.Path == "" || url.Path == "/" {
+		return "", fmt.Errorf("MakeURI(%q): missing path from URL", cloneURL)
+	} else if url.Host == "" && !strings.Contains(strings.Split(url.Path, "/")[0], ".") {
+		// We ensure our Path doesn't look like the output of TryMakeURI
+		// so that the output of this function is a fixed point.
+		// ie TryMakeURI("github.com/user/repo") == ("github.com/user/repo", nil),
+		// not an error.
+		return "", fmt.Errorf("MakeURI(%q): missing host from URL", cloneURL)
```

Comment:
These errors may be displayed to the user. If the user is not familiar with the code (or doesn't have access to it--e.g., they only have a binary), then the `MakeURI(%q):` prefix will be confusing and not-helpful. I would suggest just using error messages like `missing host from URL (%q)` (or `determining URI from repo clone URL failed: missing host from URL (%q)`) here and just passing through the error on line 35.